python/pyspark/sql/connect/typing/__init__.pyi

```diff
@@ -0,0 +1,87 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Callable, Union
+from pyspark.sql.connect.column import ExpressionOrString, ScalarFunctionExpression, ColumnOrString
+from pyspark.sql.connect.function_builder import UserDefinedFunction
+
+FunctionBuilderCallables = Union[
```

Comment:
As long as we don't care about specific arity, we should be able to replace this with

```python
class FunctionBuilderCallables(Protocol):
    def __call__(self, *exprs: ExpressionOrString) -> ScalarFunctionExpression: ...
```

I'd also keep the name singular\.

Reply:
I'll have a look. Thanks.

Reply:
We use brutal enumeration for [Pandas UDFs](https://github.com/apache/spark/blob/master/python/pyspark/sql/pandas/_typing/__init__.pyi) but this is because we want to capture arity and argument types relationships. It's arguably useful, but hard to maintain.

Reply:
Cool, I learned something again! Will follow the proposal.